src/components/NewsletterSignup/NewsletterSignup.mdx

```diff
@@ -0,0 +1,124 @@
+import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";
+
+import * as NewsletterSignupStories from "./NewsletterSignup.stories";
+import Link from "../Link/Link";
+
+<Meta of={NewsletterSignupStories} />
+
+# NewsletterSignup
+
+| Component Version | DS Version   |
+| ----------------- | ------------ |
+| Added             | `Prerelease` |
+| Latest            | `Prerelease` |
+
+## Table of Contents
+
+- {<Link href="#overview" target="_self">Overview</Link>}
+- {<Link href="#component-props" target="_self">Component Props</Link>}
+- {<Link href="#accessibility" target="_self">Accessibility</Link>}
+
+## Overview
+
+<Description of={NewsletterSignupStories} />
+
+**Note**: The `NewsletterSignup` example
+below alternately renders the "confirmation" and "error" screens on each form
+submission. This is just to demonstrate the different states of the component.
+In practice, the consuming app is responsible for handling the form submission.
+
+## Component Props
+
+<Canvas of={NewsletterSignupStories.WithControls} />
+
+<ArgTypes of={NewsletterSignupStories.WithControls} />
+
+## Accessibility
+
+The `NewsletterSignup` component is a complex component built from various Reservoir
+DS and Chakra components. The two main components are the DS `Button` component
```

Comment:
The last line should be updated to reflect "two main components". The other can be `TextInput`.

Reply:
Actually, this may be a remnant of the FeedbackBox component. The `Drawer` component is not used in this one.

Reply:
Yup I have to admit it only moved the text that was there up to fix the order of the doc without reading it. I rewrote it now and would love some more eyes on the new text. Thanks for pointing this out.